src/plugins/intel_cpu/src/emitters/snippets/aarch64/kernel_executors/gemm.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "common/utils.hpp"
+#include "cpu_memory.h"
+#include "emitters/snippets/brgemm_generic.hpp"
+#include "emitters/utils.hpp"
+#include "kai/ukernels/matmul/matmul_clamp_f32_f32_f32p/kai_matmul_clamp_f32_f32_f32p8x1biasf32_6x8x4_neon_mla.h"
+#include "kai/ukernels/matmul/matmul_clamp_f32_f32_f32p/kai_matmul_clamp_f32_f32_f32p_interface.h"
+#include "kai/ukernels/matmul/pack/kai_rhs_pack_kxn_f32p8x1biasf32_f32_f32_neon.h"
+
+namespace ov::intel_cpu::aarch64 {
+
+struct GemmKernelKaiConfig : public BrgemmGenericKernelConfig {
+public:
+    GemmKernelKaiConfig() = default;
+
+    std::unique_ptr<snippets::KernelExecutorBase::GenericConfig> get_clone_ptr() const override {
+        return std::make_unique<GemmKernelKaiConfig>(*this);
+    }
+
+    size_t hash() const override {
+        return m_hash;
+    }
+
+private:
+    size_t m_hash{SIZE_MAX};
+};
+
+class GemmKaiKernelExecutor
+    : public snippets::KernelExecutor<GemmKernelKaiConfig, kai_matmul_clamp_f32_f32_f32p_ukernel> {
+public:
+    GemmKaiKernelExecutor(GemmKernelKaiConfig config);
+
+    // No need kernel update, just update config is enough for update. The universal ukernel is reused with any config.
+    void update_kernel(const GemmKernelKaiConfig& config,
+                       std::shared_ptr<kai_matmul_clamp_f32_f32_f32p_ukernel>& kernel) const override final {
+        kernel = std::make_shared<kai_matmul_clamp_f32_f32_f32p_ukernel>(ukernel);
+    }
```

Comment:
I noticed that we don't recompile kleidiai gemm kernel for new shapes. And due to this fact you inherit kernel executor class from snippets kernel executor - not from CPU kernel executor which [supports kernel caching](https://github.com/chenhu-wang/openvino/blob/ac3469bb5f3f3f2319067eb7ebed150b04ded7b5/src/plugins/intel_cpu/src/emitters/snippets/cpu_kernel_executor_table.hpp#L19). Good idea! 👍🏼 

Probably we can even to init `kernel` in constructor? And then the method `update_kernel` will do nothing? To avoid extra initialization in each time when we call `update_by_config` or `update_by_expr`. Or we should do it anyway?

Reply:
Good idea. As `update_kernel` is the only method that update m_kernel. If it do nothing and update it in derived class, I need move `m_kernel` from `private `to `protect `to update(initialization) it directly in derived class(GemmKaiKernelExecutor) Ctor. Thanks Alexandra!